src/chainstate/stacks/boot/bns.clar

```diff
@@ -0,0 +1,973 @@
+        (err ERR_NAME_OPERATION_UNAUTHORIZED))
+      ;; The name must not be in the renewal grace period
+      (asserts!
+        (is-eq (unwrap! (is-name-in-grace-period? namespace name) (err ERR_PANIC)) false)
+        (err ERR_NAME_GRACE_PERIOD))
+      ;; The name must not be expired
+      (asserts!
+        (is-eq (unwrap! (is-name-lease-expired? namespace name) (err ERR_PANIC)) false)
+        (err ERR_NAME_EXPIRED))
+      ;; The name must not be revoked
+      (asserts!
+        (is-none (get revoked-at name-props))
+        (err ERR_NAME_REVOKED))
+      ;; The new owner does not own a name
+      (asserts!
+        can-new-owner-get-name
+        (err ERR_PRINCIPAL_ALREADY_ASSOCIATED))
+      ;; Transfer the name
+      (unwrap!
+        (update-name-ownership? namespace name contract-caller new-owner)
+        (err ERR_NAME_TRANSFER_FAILED))
+      ;; Update or clear the zonefile
+      (update-zonefile-and-props
+          namespace 
+          name  
+          (get registered-at name-props)
+          (get imported-at name-props)
+          none
+          (if (is-none zonefile-hash)
+            0x
+            (unwrap! zonefile-hash (err ERR_PANIC))))
+      (ok true))))
+
+;; NAME_REVOKE
+;; A NAME_REVOKE transaction makes a name unresolvable. The BNS consensus rules stipulate that once a name 
+;; is revoked, no one can change its public key hash or its zone file hash. 
+;; The name's zone file hash is set to null to prevent it from resolving.
+;; You should only do this if your private key is compromised, or if you want to render your name unusable for whatever reason.
+(define-public (name-revoke (namespace (buff 19))
+                            (name (buff 16)))
+  (let (
+    (owner (unwrap!
+      (nft-get-owner? names (tuple (name name) (namespace namespace)))
+      (err ERR_NAME_NOT_FOUND))) ;; The name must exist
+    (namespace-props (unwrap!
+      (map-get? namespaces ((namespace namespace)))
+      (err ERR_NAMESPACE_NOT_FOUND)))
+    (name-props (unwrap!
+      (map-get? name-properties ((name name) (namespace namespace)))
+      (err ERR_NAME_NOT_FOUND)))) ;; The name must exist
+    ;; The namespace must be launched
+    (asserts!
+      (is-some (get launched-at namespace-props))
+      (err ERR_NAMESPACE_NOT_LAUNCHED))
+    ;; The sender must match the name's current owner
+    (asserts!
+      (is-eq owner contract-caller)
+      (err ERR_NAME_OPERATION_UNAUTHORIZED))
+    ;; The name must not be expired
+    (asserts!
+      (is-eq (unwrap! (is-name-lease-expired? namespace name) (err ERR_PANIC)) false)
+      (err ERR_NAME_EXPIRED))
+    ;; The name must not be in the renewal grace period
+    (asserts!
+      (is-eq (unwrap! (is-name-in-grace-period? namespace name) (err ERR_PANIC)) false)
+      (err ERR_NAME_GRACE_PERIOD))
+    ;; The name must not be revoked
+    (asserts!
+      (is-none (get revoked-at name-props))
+      (err ERR_NAME_REVOKED))
+    ;; Clear the zonefile
```

Comment:
I think you forgot to also mark the name as revoked?

Reply:
The name is revoked by passing `(some block-height)` (line 813)